crates/store/src/db/sql/mod.rs

```diff
@@ -1234,6 +1275,7 @@ pub fn apply_block(
     // Note: ordering here is important as the relevant tables have FK dependencies.
     count += insert_block_header(transaction, block_header)?;
     count += upsert_accounts(transaction, accounts, block_header.block_num())?;
+    count += insert_scripts(transaction, &notes.iter().map(|(note, _)| note).collect::<Vec<_>>())?;
```

Comment:
We can avoid the `collect` call here if `insert_scripts` takes an iterator instead